### ISSUE TITLE
Create OpenAPI specification (Swagger) for REST API of component reconcilers

### ISSUE THREAD

**Issue by tobiscr** [kyma-incubator/reconciler#117]
**Description**

The component reconciler REST API is not described as OpenAPI specification (Swagger spec). This has to be changed and the exposed REST API and related models should to be generated on base of the specification.

The pattern of supporting different API versions in parallel should not be removed: means the URL should still include an indicator of the used API version (e.g. https://host/v1/...).

Please consider #109 before implementing this feature as it is also impacting the REST API implementation and have tickets to be aligned.

AC:
* Describe the REST API based using OpenAPI specification (Swagger spec)
* Introduce code-generators into the build process to generate models (and if meaningful also boilerplate- and middleware-code) for the exposed REST API

**Reasons**

Establishing a OpenAPI specification makes the consumption easier for API by clients, adds code-generator capabilities for REST-API models and middleware code and can simplify discussions about API changes.

**Attachments**

**Comment by tobiscr** [comment]
To be verified: already done with https://github.com/kyma-incubator/reconciler/blob/main/openapi/internal_api.yaml ?
